src/main/java/frc/robot/commands/TurretCommands.kt

```diff
@@ -0,0 +1,149 @@
+package frc.robot.commands
+
+import com.github.chen0040.glm.solvers.Coefficients
+import edu.wpi.first.math.geometry.Rotation2d
+import edu.wpi.first.math.geometry.Translation2d
+import edu.wpi.first.wpilibj.XboxController
+import edu.wpi.first.wpilibj2.command.CommandBase
+import frc.robot.subsystems.targetvision.TargetVision
+import frc.robot.subsystems.targetvision.TargetVision.Sample
+import frc.robot.subsystems.turret.Turret
+import frc.robot.utils.QuadraticPolynomial
+import org.ejml.simple.SimpleMatrix
+import java.util.function.DoubleSupplier
+import kotlin.math.abs
+import kotlin.math.atan2
+import kotlin.math.pow
+import kotlin.math.sqrt
+
+class ControlWithJoystick(private val joystickX: DoubleSupplier, private val joystickY: DoubleSupplier ) : CommandBase(){
+
+    override fun execute() {
+        val angle = atan2(joystickY.asDouble, joystickX.asDouble)
+        Turret.setTarget(Rotation2d(angle))
+    }
+
+}
+
+class TrackPrimary() : CommandBase(){
+    init {
+        addRequirements(Turret)
+    }
+
+    override fun execute() {
+        if(TargetVision.hasTargets) {
+
+            Turret.setTarget(Turret.angleToChassis.plus(Rotation2d.fromDegrees(TargetVision.primaryTarget.tx)))
+        }
+
+    }
+}
+
+//threshold to stop newton raphson method, in milliseconds
+internal const val THRESHOLD = 0.001
+internal const val INITIAL_SEED = 4.0
+
+
+
+class AimAtTarget() : CommandBase(){
+
+    init {
+        addRequirements(Turret)
+    }
+
+
+
+    private val timeOfFlight = QuadraticPolynomial(0.0, 0.0, 0.0)
+
+
+    private val euclideanNorm: (x: Double, y: Double) -> Double =
+        {x,y -> sqrt(x.pow(2) + y.pow(2))}
+
+    private fun euclideanNormDerivative(vx: QuadraticPolynomial, vy: QuadraticPolynomial, x: QuadraticPolynomial, y: QuadraticPolynomial): (Double) -> Double
+    {
+        return {t: Double  -> ( (vx.of(t) + vy.of(t)) / (2*sqrt( x.of(t).pow(2) + y.of(t).pow(2) ) ) ) }
+    }
+
+
+
+    //iteratively find roots of function by offsetting a seed based on x intercept of tangent line of last guess
+    //TODO account for chaotic fractal thingie
+    private fun newtonRaphson(func: (Double) -> Double, funcDerivative: (Double) -> Double ): Double{
+
+        val seed = INITIAL_SEED
+        var nextSeed = seed
+
+        var h = func(nextSeed)/funcDerivative(nextSeed)
+
+        while(abs(h) >= THRESHOLD){
+
+            h = func(nextSeed)/funcDerivative(nextSeed)
+            nextSeed -= h
+
+        }
+        return nextSeed
+    }
+
+
+
+    private fun getRotationalOffset(): Rotation2d{
```

Comment:
This is a kinda confusing name imo.
```suggestion
    private fun findTargetRotation(): Rotation2d {
```